packages/validate-icu-locales/src/index.ts

```diff
@@ -0,0 +1,86 @@
+import { parse } from '@formatjs/icu-messageformat-parser'
+import { info } from 'console'
+import fs from 'fs'
+import { globby } from 'globby'
+import { importFromStringSync } from 'module-from-string'
+
+const args = process.argv.slice(2)
+const pattern = args[0]
+
+const { error, table } = console
+
+type Err = {
+  err: unknown
+  value: string
+  key: string
+  filePath: string
+}
+const errors: Err[] = []
+
+const findICUError = (locales: { [key: string]: string }, filePath: string) => {
+  Object.keys(locales).forEach(key => {
+    const value = locales[key]
+
+    try {
+      parse(value)
+    } catch (err) {
+      errors.push({
+        err,
+        value,
+        key,
+        filePath,
+      })
+    }
+  })
+}
+
+const readFiles = (files: string[]) =>
+  files.forEach(file => {
+    const extension = file.split('.').pop()
+
+    if (extension === 'json') {
+      const json = fs.readFileSync(file).toString()
+      try {
+        const locales = JSON.parse(json) as Record<string, string>
+
+        findICUError(locales, file)
+      } catch (err) {
+        error(err)
+      }
+    }
+
+    if (extension === 'ts' || extension === 'js') {
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+      const d: { default: Record<string, string> } = importFromStringSync(
+        fs.readFileSync(file).toString(),
+        { transformOptions: { loader: 'ts' } },
+      )
+
+      if (typeof d.default === 'object') {
```

Comment:
I don't think the implicit cast and eslint-disable are necessary here.

```suggestion
      const mod = importFromStringSync(
        fs.readFileSync(file).toString(),
        { transformOptions: { loader: 'ts' } },
      )

      if ('default' in mod && typeof mod.default === 'object') {
```